leader-election-impl/src/main/java/com/palantir/leader/LeadershipEvents.java

```diff
@@ -58,7 +61,7 @@
         leaderPingFailure = metrics.meter(withName("leadership.ping-leader.failure", safeTags));
         leaderPingTimeout = metrics.meter(withName("leadership.ping-leader.timeout", safeTags));
         leaderPingReturnedFalse = metrics.meter(withName("leadership.ping-leader.returned-false", safeTags));
-        this.contextArgs = contextArgs.toArray(new Object[0]);
+        this.contextArgs = safeLoggingArgs.toArray(new Object[0]);
     }
 
     void proposedLeadershipFor(long round) {
```

Comment:
This logging also feels rather cumbersome: it would be great to use MDC for this. The only problem is we can't assume anything about the logging framework... :(

Reply:
yeah, it would